### ISSUE TITLE
Make xPSDesiredStateConfiguration a High Quality Resource Module

### ISSUE THREAD

**Issue by kwirkykat** [PowerShell/xPSDesiredStateConfiguration#160]
xPSDesiredStateConfiguration has been chosen as one of the two modules in the DSC Resource Kit to be updated to an experimental High Quality Resource Module. 'High Quality' here means that you would be willing to use this module in production.

We would love to hear feedback about what you think makes a resource module 'High Quality'.

**Comment by ArieHein** [comment]
For start, high quality for me is high performance which means using best practices of PowerShell programming to achieve performance. This means when you write your code, think of someone using a DSC script that has 200 registry key changes, 200 files, 200 user  creation etc and then see how your code reacts to such an amount of items. That is why I initially asked the swap to Switch statements instead of IF. And its not just in the MSFT_xRegistryResoure but rather cross the board. Thanks for putting the effort into making this more production friendly !

**Comment by kwirkykat** [comment]
@ArieHein I agree that the resources should all be scalable and perform as best they can. 

We can add performance tests for each resource to make sure they complete within a reasonable amount of time.

We could also add some scalability tests that wouldn't be able to run on each pull request Appveyor run but that perhaps would run more intermittently to test that the resources can handle large scripts.

I'll look at adding these tests to the plan.

**Comment by Dan1el42** [comment]
@kwirkykat I wonder if it is worth to put any effort into fixing PSSA issues for xFileUpload (a composite resource) and xRemoteFile if the PowerShell team is going to integrate the functionality into the File resource as per the HQRM plan.

**Comment by kwirkykat** [comment]
@Dan1el42 You're right. There's not much point in updating these resources right now when they aren't going into the HQRM anyway.

**Comment by PlagueHO** [comment]
Hi @kwirkykat - as I'm working on issues #210, #212, #213 for xServiceResource, I'm actually bringing it up to HQRM at the same time. I'm hoping to submit this tonight or tomorrow- the unit tests are taking a while (trying to get 90% coverage).

**Comment by kwirkykat** [comment]
@PlagueHO Thanks! :+1:
